ppb_vector/vector2.py

```diff
@@ -34,6 +35,37 @@ def _mkvector(value, *, castto=_fakevector):
         raise ValueError(f"Cannot use {value} as a vector-like")
 
 
+@functools.lru_cache()
+def _find_lowest_type(left: type, right: type) -> type:
```

Comment:
Should probably use typing.Type here.

Reply:
These are documented as equivalent:

> `Type[Any]` is equivalent to `Type` which in turn is equivalent to `type`, which is the root of Python’s metaclass hierarchy.

But I'll do it because metaprogramming like this can be weird.